app/src/main/res/layout/password_strength_meter.xml

```diff
@@ -0,0 +1,135 @@
+<?xml version="1.0" encoding="utf-8"?>
+<android.support.constraint.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:animateLayoutChanges="true">
+
+    <TextView
+        android:id="@+id/txtPasswordWarning"
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        android:background="@color/password_strength_medium"
+        android:padding="5dp"
+        android:text="@string/short_password_warning"
+        android:visibility="gone"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
+
+    <ProgressBar
+        android:id="@+id/progressPasswordStrength"
+        style="@style/Widget.AppCompat.ProgressBar.Horizontal"
+        android:layout_width="0dp"
+        android:layout_height="5dp"
+        android:max="9"
+        app:layout_constraintBottom_toTopOf="@+id/txtPasswordStrength"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/txtPasswordWarning" />
+
+    <TextView
+        android:id="@+id/txtPasswordStrength"
```

Comment:
Not sure that we require some text to be all caps but if we want that design then I think it's more reasonable to add it to the text field instead of the requiring all translators to do this work.

android:textAllCaps="true"

Reply:
I fully agree. Will be fixed in my next commit as well.